src/Illuminate/Database/Query/JoinClause.php

```diff
@@ -84,7 +84,7 @@ public function __construct(Builder $parentQuery, $type, $table)
      *
      * @param  \Closure|string  $first
      * @param  string|null  $operator
-     * @param  string|null  $second
+     * @param  \Illuminate\Database\Query\Expression|string|null  $second
```

Comment:
I don't follow why just the second param, and why only this method?

Reply:
It's the one I use and thus verify/vouch for